### DISCUSSION TITLE
Looking up ambiguous BTF types

### DISCUSSION THREAD

**Discussion by Ghostbaby** [cilium/ebpf#1594] [Q&A] [answered]
- code
```go
	structTypes, err := spec.AnyTypesName(struct_name)
	if err != nil {
		return 0, fmt.Errorf("无法找到结构体 %s: %v", struct_name, err)
	}
```
- log

```bash
iov_iter: found multiple types: [Struct:"iov_iter"[fields=5] Struct:"iov_iter"[fields=5]]
```

**Comment by ti-mo** [comment, marked as answer]
Not sure what this snippet means, but `AnyTypesName` is not a method on btf.Spec.

`AnyTypeByName` will only return an unambiguous type. `AnyTypesByName` will return multiple types with the same name. It's up to you to pick the one you need.
